### DISCUSSION TITLE
AI-Writer to enhance its functionality

### DISCUSSION THREAD

**Discussion by amdd252** [ALwrity/ALwrity#121] [Ideas]
## Additional Tools for AI-Writer

Here’s a list of additional tools integrated:

- [SerpScrap](https://github.com/ecoron/SerpScrap): A tool for scraping Google search results.
- [Google Rank Tracker](https://github.com/sundios/Google-rank-tracker): Track the ranking of websites on Google search.
- [Google Indexing Script](https://github.com/goenning/google-indexing-script): Script to submit URLs to Google for indexing.
- [Scylla](https://github.com/imWildCat/scylla): A free, open-source proxy pool for scraping.
- [Search Engine Parser](https://github.com/bisohns/search-engine-parser): Parse search results from Google, Bing, Yahoo, and more.
- [Trafilatura](https://github.com/adbar/trafilatura): A tool to scrape and extract text from web pages, including PDFs and articles.
- [AutoScraper](https://github.com/alirezamika/autoscraper): A smart, automatic web scraper to extract relevant data.
- [Web Scraper](https://github.com/danhilse/web-scraper): A command-line tool to scrape and organize website data.
- [KeywordKings](https://github.com/MachineKillin/KeywordKings): A keyword scraper that supports multiple search engines.

**Comment by AJaySi** [comment]
@amdd252 
Thanks a ton, once again.

@DikshaDisciplines @umesh070 
Please have a look at what we need to include soon and open issues.

1). [SerpScrap](https://github.com/ecoron/SerpScrap): A tool for scraping Google search results.
**Not applicable**, serper.dev, you.com are integrated. SerpScrap is not maintained and scraping without proxy.

2). [Trafilatura](https://github.com/adbar/trafilatura): A tool to scrape and extract text from web pages, including PDFs and articles.
Not applicable, firecrawl seems a better choice.

3). Google Indexing Script : https://github.com/AJaySi/AI-Writer/issues/122

4). scylla : ALwrity has been trying to avoid libs that are hard to install for non tech. But, we will integrate this in web app.
https://github.com/AJaySi/AI-Writer/issues/123

5). [Search Engine Parser](https://github.com/bisohns/search-engine-parser): Not maintained, serper.dev, you.com provide this functionality.

6). syclla seems a better option than AutoScraper.

7). [Web Scraper](https://github.com/danhilse/web-scraper): When we scrape from local PC without proxy, there are high chances of getting banned, scraping is almost illegal. So, scrapy, firecrawl will keep our users safe from getting banned. Same problem with KeywordKings

Note: We will have to integrate google search console APIs, the steps of getting those are difficult for non tech audience. We are also looking at free/cheap cloud APIs for keyword research. Let us know.
